R/PipeOpTaskPreproc.R

```diff
@@ -247,10 +249,21 @@ PipeOpTaskPreproc = R6Class("PipeOpTaskPreproc",
 
     predict_dt = function(dt, levels) stop("Abstract."),
 
-    select_cols = function(task) task$feature_names
+    select_cols = function(task) {
+      task$feature_types[get("type") %in% self$feature_types, get("id")]
+    }
+  ),
+  active = list(
+    feature_types = function(types) {
+      if (!missing(types)) {
+        private$.feature_types = assert_subset(types, mlr_reflections$task_feature_types)
```

Comment:
The user should not change feature_types (if he wants to restrict action, it should be with the 'affect_columns' parameter), so if (!missing) should ideally throw an error.